zerver/tests/test_user_groups.py

```diff
@@ -425,13 +425,6 @@ def test_user_group_get(self) -> None:
             response_dict["user_groups"], UserGroup.objects.filter(realm=user_profile.realm).count()
         )
 
-    def test_can_edit_user_groups(self) -> None:
-        def validation_func(user_profile: UserProfile) -> bool:
-            user_profile.refresh_from_db()
-            return user_profile.can_edit_user_groups()
-
-        self.check_has_permission_policies("user_group_edit_policy", validation_func)
-
```

Comment:
I think its good to remove this test as now we also have a group level setting. I also have a question for the cases that should be added for testing editing permissions. I think  we need to make changes in `test_user_group_edit_policy_for_creating_and_deleting_user_group`, `test_user_group_edit_policy_for_updating_user_groups`,  and `test_user_group_edit_policy_for_updating_members`. Currently in these tests we test for all possible values of `user_group_edit_policy` field. I think now for each value of `user_group_edit_policy` we should test by having value of `can_manage_group` as everyone system group and nobody system group. Tagging you @sahil839 as I think you can answer whether we need to test any other scenario too.

Reply:
Also I am not sure whether we need to separate test of testing permission for creating and deleting user groups. I think we should separate them as now we all make use of group  level setting in determining whether a user group can be deleted by user or not.

Reply:
```
I think now for each value of user_group_edit_policy we should test by having value of can_manage_group as everyone system group and nobody system group.
```
It is not clear that tests for which settings you are referring to here. Are you referring to test the realm level setting or the group level setting.

Reply:
> Also I am not sure whether we need to separate test of permissions for creating and deleting user groups. I think we should separate them as now we also make use of group level setting in determining whether a user group can be deleted by user or not and we would be also adding a separate setting for configuring user group creation permissions soon.

Is it decided that we would have a group-level setting for deleting groups? I cannot see in any of the issues that I have looked at. And it's not clear to me whether it will be a separate setting or "Who can manage that group" setting.

Reply:
> > I think now for each value of user_group_edit_policy we should test by having value of can_manage_group as everyone system group and nobody system group.
> 
> It is not clear that tests for which settings you are referring to here. Are you referring to test the realm level setting or the group level setting.

Do we need separate test for both settings? I was thinking to test both realm and group level settings in a single test beacuse to decide whether user has permissions to update user group will be determined both by group and realm level setting.

Reply:
> > Also I am not sure whether we need to separate test of permissions for creating and deleting user groups. I think we should separate them as now we also make use of group level setting in determining whether a user group can be deleted by user or not and we would be also adding a separate setting for configuring user group creation permissions soon.
> 
> Is it decided that we would have a group-level setting for deleting groups? I cannot see in any of the issues that I have looked at. And it's not clear to me whether it will be a separate setting or "Who can manage that group" setting.

No, there is no issue which mentions adding a separate setting for configuring user group deletion permissions. From my understanding and work done in this PR till now I have assumed that user can delete the group if he has the permissions to manage the group.

Reply:
> Do we need separate test for both settings? I was thinking to test both realm and group level settings in a single test beacuse to decide whether user has permissions to update user group will be determined both by group and realm level setting.

The realm setting will be "Who can manage all groups" and this would override the group level permission I guess. So, would be better to have separate tests I guess. One for the realm setting where we can include tests to make sure that the group-level permissions are overridden correctly. And one for the group level setting, where realm level setting would not matter. 

I am not sure whether we should test all the system groups or not. But would definitely be helpful to include a couple of role-based groups like "admins" and "moderators" and not just "everyone" and "nobody" groups.

Reply:
> No, there is no issue which mentions adding a separate setting for configuring user group deletion permissions. From my understanding and work done in this PR till now I have assumed that user can delete the group if he has the permissions to manage the group.

I am not sure whether we would want all the users who are allowed to add/remove the members to delete the group. Would be good to discuss this on CZO first before moving forward.

Reply:
As discussed [here](https://chat.zulip.org/#narrow/stream/101-design/topic/New.20permissions.20model/near/1625159) there is no need to add a separate setting for group deletion permissions. I think we should now separate `test_user_group_edit_policy_for_creating_and_deleting_user_group` into `test_user_group_edit_policy_for_creating_user_group`, `test_realm_level_setting_for_deleting_user_group` and `test_group_level_setting_for_deleting_user_group`. @sahil839 FYI.

Reply:
Yes, sounds good.